File: app/src/main/java/me/knighthat/interactivedeck/component/Buttons.java
package me.knighthat.interactivedeck.component;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.Collection;
import java.util.Collections;
import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

import me.knighthat.interactivedeck.component.ibutton.IButton;

public class Buttons {

    private static final @NotNull Map<UUID, IButton> BUTTONS = new HashMap<>();

    public static @NotNull Collection<IButton> list() {
        return Collections.unmodifiableCollection(BUTTONS.values());
    }

    public static void add( @NotNull IButton button ) {
        BUTTONS.put(button.uuid(), button);
    }

    public static void clear() {
        BUTTONS.clear();
    }

    public static @Nullable IButton get( @NotNull UUID uuid ) {
        return BUTTONS.getOrDefault(uuid, null);
    }

    public static @Nullable IButton get( @NotNull String id ) {
        return get(UUID.fromString(id));
    }

    public static void remove( @NotNull UUID uuid ) {
        BUTTONS.remove(uuid);
    }

    public static void remove( @NotNull String id ) {
        remove(UUID.fromString(id));
    }
}

File: app/src/main/java/me/knighthat/interactivedeck/button/Buttons.kt
package me.knighthat.interactivedeck.button

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import me.knighthat.interactivedeck.component.ibutton.IButton

class Buttons : ViewModel() {

    private val _buttons = MutableLiveData<MutableList<IButton>>()
    val buttons: LiveData<MutableList<IButton>> get() = _buttons

    fun set(buttons: MutableList<IButton>) {
        _buttons.value = buttons
    }
}